frontend/src/features/documents/list/DocumentListView.tsx

```diff
@@ -0,0 +1,73 @@
+import { TableSort } from 'components/Table/TableSort';
+import { DocumentTypes } from 'constants/documentTypes';
+import { defaultDocumentFilter, IDocumentFilter } from 'interfaces/IDocumentResults';
+import { orderBy } from 'lodash';
+import { Api_Document } from 'models/api/Document';
+import React from 'react';
+
+import { DocumentFilterForm } from './DocumentFilter/DocumentFilterForm';
+import { DocumentResults } from './DocumentResults/DocumentResults';
+import * as Styled from './styles';
+
+export interface IDocumentListViewProps {
+  entityId: number;
+  documentType: DocumentTypes;
+  isLoading: boolean;
+  documentResults: Api_Document[];
+  hideFilters?: boolean;
+  defaultFilters?: IDocumentFilter;
+}
+/**
+ * Page that displays documents information.
+ */
+export const DocumentListView: React.FunctionComponent<IDocumentListViewProps> = (
+  props: IDocumentListViewProps,
+) => {
+  const { documentResults, isLoading, defaultFilters, hideFilters } = props;
+  const [sort, setSort] = React.useState<TableSort<Api_Document>>({});
+
+  const [filters, setFilters] = React.useState<IDocumentFilter>(
+    defaultFilters ?? defaultDocumentFilter,
+  );
+
+  const sortedFilteredDocuments = React.useMemo(() => {
+    if (documentResults?.length) {
+      let documentItems = [...documentResults];
+
+      if (filters) {
+        documentItems = documentItems.filter(document => {
+          return (
+            (!filters.documentType || document.documentTypeId === filters.documentType) &&
+            (!filters.status || document.statusCode === filters.status)
+          );
+        });
+      }
+      if (sort) {
+        const sortFields = Object.keys(sort);
+        if (sortFields?.length > 0) {
+          const keyName = (sort as any)[sortFields[0]];
+          return orderBy(documentItems, sortFields[0], keyName) as Api_Document[];
```

Comment:
q: Why is the type being cohersed here, is the orderBy changing the type?

Reply:
Done, initially when I was writing the code it was giving some type error so I did it, later on it seems it was needed